dask_cloudprovider/__init__.py

```diff
@@ -42,3 +42,9 @@ def __getattr__(name):
             "DigitalOcean cluster managers must be imported from the digitalocean subpackage. "
             f"Please import dask_cloudprovider.digitalocean.{name}"
         )
+
+    if name in ["FlyMachineCluster"]:
+        raise ImportError(
+            "Fly.io cluster managers must be imported from the fly subpackage. "
+            f"Please import dask_cloudprovider.fly.{name}"
+        )
```

Comment:
These shims only exist for historic reasons because previously some of these classes were available at the top level. I don't see a need to add a shim for a new provider.